app/code/Magento/Sales/view/adminhtml/templates/order/totals/tax.phtml

```diff
@@ -59,7 +59,10 @@ $randomHelper = $block->getData('randomHelper');
             ?>
 
             <?php foreach ($rates as $rate): ?>
-                <tr id="rate-<?= /* @noEscape */ $rate->getId() ?>"
+                <?php
+                    $rateId = $randomHelper->getRandomString(20);
```

Comment:
Could we use `$rate['code']` instead of the random string? The code is supposed to be unique.

Reply:
Sadly we cannot since `code` might not be "id syntax compatible".
Here's a `var_dump()` of one of my `$rate`:
```
array(5) {
    ["code"]=>
    string(10) "19% Steuer"
    ["title"]=>
    string(10) "19% Steuer"
    ["percent"]=>
    float(19)
    ["position"]=>
    string(1) "0"
    ["priority"]=>
    string(1) "1"
}
```

`<tr id="rate-19% Steuer"` (It's "19% tax" in German btw) would just not work.

The real solution would of course be to expand the `$rate`, but since this is a "production breaking bug" and the same code is already used here: https://github.com/magento/magento2/blob/2.4.1/app/code/Magento/Sales/view/adminhtml/templates/order/totals/tax.phtml#L91 (which would also be in need of a better fix) I came up with this quick fix, so that at least our customers are able to work again.

Reply:
It makes sense, thank you for the explanation